content/javascript/concepts/map/map.md

```diff
@@ -12,83 +12,66 @@ CatalogContent:
 
-A new map can be defined as follows. This map will store the number of fruits a store has.
+- Values mapped to existing keys can be overwritten later.
+- Keys/values can either be an [object](https://www.codecademy.com/resources/docs/javascript/objects) or a variable of [any data type](https://www.codecademy.com/resources/docs/javascript/data-types).
+- Maps are directly iterable whereas objects are not.
 
-```js
-const fruits = new Map();
-```
-
-## Storing Data in Map
+## Syntax
 
-Key-value pairs are stored in `Map` using the `.set()` method.
+A map can be defined with the `new` keyword:
 
 ```js
-fruits.set('Apples', 5);
-fruits.set('Oranges', 8);
+const map = new Map();
 ```
 
-This stores in `fruits`, 5 apples and 8 oranges.
+The example above creates an empty `Map` object, `map`.
 
-## Retrieving Data from Map
+## Example
 
-Values are retrieved by their keys from `Map` objects using the `.get()` method.
+To create a non-empty `Map` object, an array of arrays is passed into `Map()`. Each inner array represents a key-value pair:
 
 ```js
-console.log(fruits.get('Apples')); // Output: 5
-console.log(fruits.get('Oranges')); // Output: 8
-```
-
-This retrieves the information that there are 5 apples and 8 oranges in the store.
+const hogwartsStudents = new Map([
+  ['Gryffindor', 'Harry Potter'],
+  ['Slytherin', 'Draco Malfoy'],
+  ['Hufflepuff', 'Cedric Diggory'],
+]);
 
-## Determining Whether Keys are in Map
-
-Checking whether a key is in a `Map` object can be done using the `.has()` method.
-
-```js
-console.log(fruits.has('Apples')); // Output: true
-console.log(fruits.has('Bananas')); // Output: false
+console.log(hogwartsStudents);
 ```
 
-The first statement checks that there are apples in `fruits`, while the second statement finds that there are no bananas in `fruits` and returns `false`.
-
-## Retrieving the Size of a Map
-
-The number of keys in a `Map` object can be checked using the `.size` property.
+The output will be:
 
-```js
-console.log(fruits.size); // Output: 2
 ```
-
-There are 2 key-value pairs in `fruits`: `'Apples' : 5` and `'Oranges' : 8`.
-
-## Removing Data from Map
-
-Data can be removed from `Map` using the `.delete()` method.
-
-```js
-console.log(fruits.delete('Oranges')); // Output: true
-console.log(fruits.delete('Strawberries')); // Output: false
+Map(3) {
+  'Gryffindor' => 'Harry Potter',
+  'Slytherin' => 'Draco Malfoy',
+  'Hufflepuff' => 'Cedric Diggory'
+}
 ```
 
-The first statement returns `true` because `fruits` contain `Oranges` as a key and has removed it.
-The second statement returns `false` because `Strawberries` is not a key in fruits.
-
 ## Codebyte Example
 
+In the example below, an `addressBook` maps a person's name to a phone number. By the end of the program:
+
+- Tom has moved and deleted their phone number with `.delete()`.
+- Paul has changed their phone number using `.set()`.
```

Comment:
I think .delete() and .set() should link to the new entries for them

Reply:
Definitely will do that once the links live and copy/paste-able. 😃